shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -469,7 +469,19 @@
 bool FlutterPlatformViewsController::SubmitFrame(GrDirectContext* gr_context,
                                                  std::shared_ptr<IOSContext> ios_context,
                                                  std::unique_ptr<SurfaceFrame> frame) {
-  FML_DCHECK(flutter_view_);
+  // Warning: flutter_view_ could be deleted on the platform thread between load
+  // and retain.  Both are atomic but their composition isn't.
+  fml::scoped_nsobject<UIView> flutter_view([flutter_view_.load() retain]);
```

Comment:
why is `retain` required again?

Reply:
Because we are dealing with multiple threads.  The platform thread could delete the view while we are running SubmitFrame on the raster thread.  This protects us more.

Reply:
interesting. Do you know what event can cause the platform thread to delete the view?  I guess in add-to-app that can happen at any time.